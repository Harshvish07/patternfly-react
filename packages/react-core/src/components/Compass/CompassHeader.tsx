import styles from '@patternfly/react-styles/css/components/Compass/compass';
import { css } from '@patternfly/react-styles';

interface CompassHeaderProps {
  /** Content of the logo area */
  logo?: React.ReactNode;
  /** Content of the navigation area */
  nav?: React.ReactNode;
  /** Content of the profile area */
  profile?: React.ReactNode;
}

export const CompassHeader: React.FunctionComponent<CompassHeaderProps> = ({ logo, nav, profile }) => (
  <>
    <div className={css(`${styles.compass}__logo`)}>{logo}</div>
    <div className={css(styles.compassNav)}>{nav}</div>
    <div className={css(styles.compassProfile)}>{profile}</div>
  </>
);

CompassHeader.displayName = 'CompassHeader';
