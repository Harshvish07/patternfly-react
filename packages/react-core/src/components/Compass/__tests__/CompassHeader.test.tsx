import { render, screen } from '@testing-library/react';
import { CompassHeader } from '../CompassHeader';
import styles from '@patternfly/react-styles/css/components/Compass/compass';

test('Renders without children', () => {
  render(
    <div data-testid="test-header">
      <CompassHeader />
    </div>
  );
  expect(screen.getByTestId('test-header').firstChild).toBeVisible();
});

test('Renders logo content when provided', () => {
  render(<CompassHeader logo={<div>Logo content</div>} />);
  expect(screen.getByText('Logo content')).toBeVisible();
});

test('Renders nav content when provided', () => {
  render(<CompassHeader nav={<div>Nav content</div>} />);
  expect(screen.getByText('Nav content')).toBeVisible();
});

test('Renders profile content when provided', () => {
  render(<CompassHeader profile={<div>Profile content</div>} />);
  expect(screen.getByText('Profile content')).toBeVisible();
});

test('Renders all content when all props are provided', () => {
  render(<CompassHeader logo={<div>Logo</div>} nav={<div>Nav</div>} profile={<div>Profile</div>} />);
  expect(screen.getByText('Logo')).toBeVisible();
  expect(screen.getByText('Nav')).toBeVisible();
  expect(screen.getByText('Profile')).toBeVisible();
});

test(`Renders logo with ${styles.compass}__logo class`, () => {
  render(<CompassHeader logo={<div>Logo</div>} />);
  expect(screen.getByText('Logo').parentElement).toHaveClass(`${styles.compass}__logo`);
});

test(`Renders nav with ${styles.compassNav} class`, () => {
  render(<CompassHeader nav={<div>Nav</div>} />);
  expect(screen.getByText('Nav').parentElement).toHaveClass(styles.compassNav);
});

test(`Renders profile with ${styles.compassProfile} class`, () => {
  render(<CompassHeader profile={<div>Profile</div>} />);
  expect(screen.getByText('Profile').parentElement).toHaveClass(styles.compassProfile);
});

test('Matches the snapshot', () => {
  const { asFragment } = render(
    <CompassHeader logo={<div>Logo</div>} nav={<div>Nav</div>} profile={<div>Profile</div>} />
  );
  expect(asFragment()).toMatchSnapshot();
});
