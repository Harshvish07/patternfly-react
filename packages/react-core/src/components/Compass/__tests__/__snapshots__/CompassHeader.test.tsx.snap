// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="pf-v6-c-compass__logo"
  >
    <div>
      Logo
    </div>
  </div>
  <div
    class="pf-v6-c-compass__nav"
  >
    <div>
      Nav
    </div>
  </div>
  <div
    class="pf-v6-c-compass__profile"
  >
    <div>
      Profile
    </div>
  </div>
</DocumentFragment>
`;
